src/librbd/Operations.cc

```diff
@@ -539,9 +539,11 @@ int Operations<I>::rename(const char *dstname) {
       return r;
     }
   } else {
-    RWLock::RLocker owner_lock(m_image_ctx.owner_lock);
     C_SaferCond cond_ctx;
-    execute_rename(dstname, &cond_ctx);
+    {
+      RWLock::RLocker owner_lock(m_image_ctx.owner_lock);
+      execute_rename(dstname, &cond_ctx);
+    }
```

Comment:
@dillaman in `execute_rename()`, `m_image_ctx.old_format` case, `RenameRequest` is sent in a different contex and needs the owner lock. From qa/workunits/rbd/cli_generic.sh run:
```
rbd create --image-format 1 -s 1 foo
rbd rename foo foo2

2017-11-15 12:31:05.183 7f847cff9700 -1 /home/mgolub/ceph/ceph.ci/src/librbd/operation/Request.cc: In function 'void librbd::operation::Request<ImageCtxT>::send() [with ImageCtxT = librbd::ImageCtx]' thread 7f847cff9700 time 2017-11-15 12:31:05.189232
/home/mgolub/ceph/ceph.ci/src/librbd/operation/Request.cc: 25: FAILED assert(image_ctx.owner_lock.is_locked())
```

Reply:
Thanks -- update pushed

Reply:
nit: the log message now looks outdated

Reply:
@dillaman I mean the commit log message

Reply:
@trociny ... tweaked